proxy/http2/Http2Stream.cc

```diff
@@ -331,8 +331,12 @@ Http2Stream::do_io_write(Continuation *c, int64_t nbytes, IOBufferReader *abuffe
   write_vio.op        = VIO::WRITE;
   response_reader     = abuffer;
 
-  update_write_request(abuffer, nbytes, false);
-
+  if (c != nullptr && nbytes > 0 && this->is_client_state_writeable()) {
+    update_write_request(abuffer, nbytes, false);
+  } else if (!this->is_client_state_writeable()) {
```

Comment:
Are you assuming when HttpTunnel::producer_run() calls this do_io_write(), the Http2Stream is closing / closed ?
In that case, we should just return `nullptr` in the beginning of this function which what `UnixNetVConnection` does, I think.

Reply:
Yes, that was my assumption, and this does look like a better solution.  I will give that one a try.